### PR TITLE
new automation cmdlets

### PR DESCRIPTION
**Pull request by neilwheeler**
> We are allowing contributions for now with limited review whilst we learn what common practices, approach and recommendations we can follow prior to promoting these modules to our customers.

### Manual Tasks Checklist
- [ ] Increment module version number in /Juriba.Dashworks/Juriba.Dashworks.psd1
- [ ] Add new functions to 'FunctionsToExport' in /Juriba.Dashworks/Juriba.Dashworks.psd1

### CODE REVIEW

File: Juriba.Dashworks/Public/New-DwDashboardHalfDonutWidget.ps1 [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            "colourTemplateId"          = 1
+            "aggregateFunctionId"       = 1
+            "maximumValues"             = 10
+            "drilldownEnabled"          = $true
+            "maxRows"                   = 0
+            "maxColumns"                = 0
+            "cardTypeIsAggregate"       = $false
+            "splitBy"                   = $SplitBy
+            "aggregateBy"               = ""
+            "orderByDescending"         = if ($OrderByDescending) { $true }else { $false }
```
Comment (dgreenshields) [inline Juriba.Dashworks/Public/New-DwDashboardHalfDonutWidget.ps1:46]: The definition of OrderByDescending is:
 
 [bool]$OrderByDescending = $false
 
 thus I don't see how you'd need more than:
 
 "orderByDescending"  = $OrderByDescending
 
 Given that the variable has a default value of false. It does the same thing.

Comment (neilwheeler) [reply Juriba.Dashworks/Public/New-DwDashboardHalfDonutWidget.ps1]: Good point, changed.

File: Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1 [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            "showLegend"                = $true
+            "orientationIsVertical"     = $false
+            "layout"                    = $null
+            "displayDataLabels"         = $false
+            "categoriseBy"              = $CategoriseBy
+            "categorisationViewType"    = "Stacked"
+            "categoriseByMaximumValues" = 10
+        } | ConvertTo-Json
+
+        $contentType = "application / json"
```
Comment (dgreenshields) [inline Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1:55]: This should be: "application/json" without the spaces as far as I can tell from:

https://datatracker.ietf.org/doc/html/rfc2045#section-5.1

File: Juriba.Dashworks/Public/New-DwDashboardCardWidget.ps1 [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            "orderByField"              = $null
+            "showLegend"                = $false
+            "orientationIsVertical"     = $false
+            "layout"                    = "IconAndText"
+            "displayDataLabels"         = $false
+            "categoriseBy"              = ""
+            "categorisationViewType"    = "None"
+            "categoriseByMaximumValues" = $null
+        } | ConvertTo-Json
+        ContentType = "application / json"
```
Comment (dgreenshields) [inline Juriba.Dashworks/Public/New-DwDashboardCardWidget.ps1:51]: This should be: "application/json" without the spaces as far as I can tell from:

https://datatracker.ietf.org/doc/html/rfc2045#section-5.1

**Review by dgreenshields — CHANGES_REQUESTED** [review]

File: Juriba.Dashworks/Public/New-DwDashboardSection.ps1 [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        [int]$Width = 2
+    )
+
+    $request = @{
+        Uri         = ($Instance + "/apiv1/dashboard/" + $DashboardId + "/section/")
+        Method      = "Post"
+        Body        = @{
+            "width" = $Width
+        } | ConvertTo-Json
+        ContentType = "application / json"
```
[the remark this answers is not in the document]
Comment (neilwheeler) [reply Juriba.Dashworks/Public/New-DwDashboardSection.ps1]: Good point. Done.

File: Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1 (hunk 2 above)
Comment (neilwheeler) [reply Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1]: Done.

File: Juriba.Dashworks/Public/New-DwList.ps1 [hunk 5]
```diff
@@ -0,0 +1,57 @@
+Function New-DwList {
```
[the remark this answers is not in the document]
Comment (neilwheeler) [reply Juriba.Dashworks/Public/New-DwList.ps1:1]: good idea. done.

File: Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1 [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            "colourTemplateId"          = 1
+            "aggregateFunctionId"       = 1
+            "maximumValues"             = 10
+            "drilldownEnabled"          = $true
+            "maxRows"                   = 0
+            "maxColumns"                = 0
+            "cardTypeIsAggregate"       = $false
+            "splitBy"                   = $SplitBy
+            "aggregateBy"               = ""
+            "orderByDescending"         = if ($OrderByDescending) { $true } else { $false }
```
Comment (dgreenshields) [inline Juriba.Dashworks/Public/New-DwDashboardBarWidget.ps1:43]: "orderByDescending"         = $OrderByDescending

File: Juriba.Dashworks/Public/New-DwDashboardTableWidget.ps1 [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            "colourTemplateId"          = 1
+            "aggregateFunctionId"       = 1
+            "maximumValues"             = 10
+            "drilldownEnabled"          = $true
+            "maxRows"                   = 0
+            "maxColumns"                = 0
+            "cardTypeIsAggregate"       = $false
+            "splitBy"                   = $SplitBy
+            "aggregateBy"               = ""
+            "orderByDescending"         = if ($OrderByDescending) { $true }else { $false }
```
Comment (dgreenshields) [inline Juriba.Dashworks/Public/New-DwDashboardTableWidget.ps1:48]: "orderByDescending"         = $OrderByDescending

**Review by dgreenshields — APPROVED** [review]

**Review by andrewrmoreland — APPROVED** [review]

**Pull request merged**